UIImageColors/Sources/UIImageColors.swift

```diff
@@ -171,7 +177,17 @@ fileprivate extension Double {
 }
 
 extension UIImage {
-    private func resizeForUIImageColors(newSize: CGSize) -> UIImage {
+    private func resizeForUIImageColors(newSize: CGSize) -> UIImage? {
+        #if os(OSX)
+        let frame = CGRect(origin: .zero, size: newSize)
```

Comment:
nit: can we add indentation in the if block?

Also, it would be nicer if we can abstract away each block into its own method